custom/sensors/SparkEncoder.java

```diff
@@ -2,85 +2,196 @@
+     *                            kDisplacement for position readings, kRate for velocity readings
+     */
+    public SparkEncoder(Spark spark, boolean reverseDirection, double distancePerRotation, PIDSourceType pidSource) {
         encoder = new CANEncoder(spark);
         this.reverseDirection = reverseDirection;
-        this.distancePerPulse = distancePerPulse;
-        setPIDSourceType(PIDSourceType.kDisplacement);
+        this.distancePerRotation = distancePerRotation;
+        setPIDSourceType(pidSource);
     }
 
+    /**
+     * Built-in encoder on CANSpark NEO
+     * Implements generic encoder interface
+     * 
+     * @param spark
+     *                            Spark NEO motor used to construct encoder
+     * @param reverseDirection
+     *                            True to reverse direction of encoder
+     * @param distancePerRotation
+     *                            Conversion factor from encoder readings to distance/velocity readings
+     *                            Should be in units of meters per second
+     */
+    public SparkEncoder(Spark spark, boolean reverseDirection, double distancePerRotation) {
+        this(spark, reverseDirection, distancePerRotation, PIDSourceType.kDisplacement);
+    }
+
+    /**
+     * Built-in encoder on CANSpark NEO
+     * Implements generic encoder interface
+     * 
+     * @param spark
+     *                            Spark NEO motor used to construct encoder
+     * @param distancePerRotation
+     *                            Conversion factor from encoder readings to distance/velocity readings
+     *                            Should be in units of meters per second
+     */
+    public SparkEncoder(Spark spark, double distancePerRotation) {
+        this(spark, false, distancePerRotation);
+    }
+
+    /**
+     * Built-in encoder on CANSpark NEO
+     * Implements generic encoder interface
+     * DistancePerRotation set to 1 to return raw encoder readings
+     * 
+     * @param spark
+     *              Spark NEO motor used to construct encoder
+     */
+    public SparkEncoder(Spark spark) {
+        this(spark, false, 1.0);
+    }
+
+    /**
+     * Set encoder type
+     * 
+     * @param pidSource
+     *                  kDisplacement for position readings, kRate for velocity readings
+     */
     @Override
     public void setPIDSourceType(PIDSourceType pidSource) {
         this.pidSource = pidSource;
     }
 
+    /**
+     * @return pidSource
+     *         returns type of encoder
+     */
     @Override
     public PIDSourceType getPIDSourceType() {
         return pidSource;
     }
 
+    /**
+     * Returns encoder reading depending on pidSource
+     */
     @Override
     public double pidGet() {
-        if (pidSource == PIDSourceType.kDisplacement) {
-            return getDistance();
-        } else {
-            return getRate();
+        try {
+            return pidGetSafely();
+        }
+        catch (InvalidSensorException e) {
```

Comment:
The `InvalidSensorException` is thrown by our implementation of CANSensor at https://github.com/RoboticsTeam4904/standard/blob/master/custom/sensors/CANSensor.java#L59.
I don't think `CANEncoder.getPosition()` should ever throw the exception, so we don't need to check for it. Unless reading from the encoder would throw a different type of error or should throw that exception, this should be removed.